### PR TITLE
Update calmly-writer from 2.0.39 to 2.0.42

### PR DESCRIPTION
**Pull request by neutric**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by neutric** [comment]
```
Version '2.0.42' differs from '' retrieved by livecheck.
```
Locally I get:
```
% brew livecheck --debug --cask calmly-writer

Cask:             calmly-writer
Livecheckable?:   Yes

URL:              https://calmlywriter.com/releases/x64/download.php
Strategy:         HeaderMatch

Matched Versions:
2.0.42
calmly-writer: 2.0.42 ==> 2.0.42
```

**Comment by bevanjkay** [comment]
@neutric I'm getting an incorrect result on `arm` so probably need to add a regex..

```
Cask:             calmly-writer
Livecheckable?:   Yes

URL:              https://calmlywriter.com/releases/arm64/download.php
Strategy:         HeaderMatch

Matched Versions:
64
calmly-writer: 2.0.39 ==> 64
```

For reference:

```
curl -I https://calmlywriter.com/releases/arm64/download.php                                           
HTTP/1.1 302 Moved Temporarily
Date: Mon, 30 May 2022 00:53:09 GMT
Server: Apache
Location: https://www.calmlywriter.com/releases/Calmly Writer-2.0.42-arm64.dmg
Content-Type: text/html
```

**Comment by neutric** [comment]
@bevanjkay Ah, thanks! Will take care of that after some 💤 ...

File: Casks/calmly-writer.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@
   livecheck do
     url "https://calmlywriter.com/releases/#{arch}/download.php"
     strategy :header_match
+    regex(/Calmly\sWriter[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
```
Comment (bevanjkay) [inline Casks/calmly-writer.rb:21]:
```suggestion
    regex(/Calmly\sWriter[._-]v?(\d+(?:\.\d+)+)[._-]#{arch}\.dmg/i)
```

Comment (bevanjkay) [reply Casks/calmly-writer.rb:21]: I think it will need the arch in the regex - for arm64 at least.

Comment (miccal) [reply Casks/calmly-writer.rb]:
```
|-> curlfollow https://calmlywriter.com/releases/arm64/download.php
Location: https://www.calmlywriter.com/releases/Calmly Writer-2.0.42-arm64.dmg
```
```
|-> curlfollow https://calmlywriter.com/releases/x64/download.php
Location: https://www.calmlywriter.com/releases/Calmly Writer-2.0.42.dmg
```

Comment (bevanjkay) [reply Casks/calmly-writer.rb]: 👍🏼 and another shell function to borrow from your personal-files repo 😆 `curlfollow`

Comment (miccal) [reply Casks/calmly-writer.rb]: Go for it :) very handy one.

**Comment by miccal** [comment]
Both `livecheck`'s are working for me:
```
|-> brew livecheck --debug calmly-writer

Cask:             calmly-writer
Livecheckable?:   Yes

URL:              https://calmlywriter.com/releases/x64/download.php
Strategy:         HeaderMatch

Matched Versions:
2.0.42
calmly-writer: 2.0.42 ==> 2.0.42
```
```
|-> brew livecheck --debug calmly-writer

Cask:             calmly-writer
Livecheckable?:   Yes

URL:              https://calmlywriter.com/releases/arm64/download.php
Strategy:         HeaderMatch

Matched Versions:
2.0.42
calmly-writer: 2.0.42 ==> 2.0.42
```

**Review by miccal — APPROVED** [review]

**Comment by neutric** [comment]
Thanks guys!

**Pull request merged**